pallets/invulnerables/src/lib.rs

```diff
@@ -142,9 +163,39 @@ pub mod pallet {
                 Error::<T>::TooManyInvulnerables
             );
 
+            let mut new_with_keys = Vec::new();
+
+            // check if the invulnerables have associated validator keys before they are set
+            for account_id in &new {
+                // don't let one unprepared collator ruin things for everyone.
+                let validator_key = T::ValidatorIdOf::convert(account_id.clone());
+                match validator_key {
+                    Some(key) => {
+                        // key is not registered
+                        if !T::ValidatorRegistration::is_registered(&key) {
+                            Self::deposit_event(Event::InvalidInvulnerableSkipped {
+                                account_id: account_id.clone(),
+                            });
+                            continue;
+                        }
+                        // else condition passes; key is registered
+                    }
+                    // key does not exist
+                    None => {
+                        Self::deposit_event(Event::InvalidInvulnerableSkipped {
+                            account_id: account_id.clone(),
+                        });
+                        continue;
+                    }
+                }
+
+                new_with_keys.push(account_id.clone());
+            }
```

Comment:
You also need to add this in the add_invulnerable extrinsic

Reply:
done